src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java

```diff
@@ -165,4 +173,28 @@ public void validateEmail(String email) throws SubscriptionValidationException {
             throw new SubscriptionValidationException("Wrong email address: " + email);
         }
     }
+
+	public void validateWithSchema(Subscription subscription) throws SubscriptionValidationException {
```

Comment:
I would call this function from the validateSubscription method in this file since it is validating the subscription using a schema.